### DISCUSSION TITLE
How to render a UINavigationController instance exposed via NativeModules from Swift?

### DISCUSSION THREAD

**Discussion by mohsinulhaq** [software-mansion/react-native-screens#858] [General]
I have a setup where I have bottom navigator created via `createBottomTabNavigator`, and each tab rendering a native stack navigator created via `createNativeStackNavigator`. Now I'm trying to merge another app in written in Swift UIkit.
That app uses `UITabBarController` to render tabs created via `UINavigationController`.
Is it possible, either at tab level or screen level to get those screens to render in React Native?

I tried exposing the `UINavigationController` stack via the `NativeModules` interface, but I'm not sure how to plug that into the JS bottom navigator.

Any help would be greatly appreciated.

**Comment by mohsinulhaq** [comment]
@WoLewicki sorry for tagging you, but is there something that can be done?

**Reply by WoLewicki** [reply]
I think you should do something similar to what is done in `react-native-screens` with `ScreenStack` and `Screen` components. It requires handling of the navigation lifecycle etc., but for sure it is possible. Look at how the `native-stack` navigator is created here: https://github.com/software-mansion/react-native-screens/blob/master/src/native-stack/navigators/createNativeStackNavigator.tsx#L80 and try to do something similar for your native component.

**Reply by mohsinulhaq** [reply]
@WoLewicki I was able to render a UIViewController inside RNSStack. But the next issue I faced was this:
https://github.com/software-mansion/react-native-screens/discussions/866
